auth/src/express/controllers/saml-controller.ts

```diff
@@ -0,0 +1,185 @@
+import { Request, Response } from 'express';
+import { Factory } from 'final-di';
+import { OnGet, OnPost, Req, Res, RestController } from 'rest-app';
+
+import { AxiosResponse } from 'axios';
+import { DatastoreAdapter } from '../../adapter/datastore-adapter';
+import { AuthHandler } from '../../api/interfaces/auth-handler';
+import { Datastore } from '../../api/interfaces/datastore';
+import { HttpHandler } from '../../api/interfaces/http-handler';
+import { AuthService } from '../../api/services/auth-service';
+import { HttpService } from '../../api/services/http-service';
+import { Logger } from '../../api/services/logger';
+import { Config } from '../../config';
+import { User } from '../../core/models/user';
+import saml from '../../saml';
+import { AuthServiceResponse } from '../../util/helper/definitions';
+import { createResponse } from '../../util/helper/functions';
+
+export interface SamlUser {
+    saml_id: string, title?: string, first_name: string, last_name: string, email?: string, gender?: string, pronoun?: string, is_active?: boolean, is_physical_person?: boolean | string
+}
+
+interface SamlBackendCall {
+    action: string, data: [SamlUser]
+}
+
+@RestController({
+    prefix: 'system/saml'
+})
+export class SamlController {
+    @Factory(AuthService)
+    private _authHandler: AuthHandler;
+
+    @Factory(DatastoreAdapter)
+    private readonly _datastore: Datastore;
+
+    @Factory(HttpService)
+    private readonly _httpHandler: HttpHandler;
+
+    /**
+     * Indicates if the service is available
+     * @returns Generic response
+     */
+    @OnGet()
+    public index(): AuthServiceResponse {
+        return createResponse({}, 'SAML SP service is available');
+    }
```

Comment:
I stumbled upon this.
GET /system/saml/index
responds with 404.
When looking at the other handlers below I suspect the response object should be taken as parameter here as well?

Maybe this is a special case and you have your reasons, just sharing my thoughts.

Reply:
Hi,
in order for the SAML paths to be resolved, the change to the proxy https://github.com/OpenSlides/OpenSlides/pull/6577 must be taken into account.
This introduces the necessary additional rule to forward SAML requests to the auth service.